psi4/src/psi4/dfmp2/mp2.cc

```diff
@@ -1158,7 +1158,7 @@ void RDFMP2::form_Pab() {
 
     // Memory
     size_t doubles = ((size_t)(options_.get_double("DFMP2_MEM_FACTOR") * memory_ / 8L));
-    doubles -= navir * navir;
+    doubles -= static_cast<unsigned long> (navir) * navir;
```

Comment:
Normally we should cast to `size_t` here and below. I would actually vote to change `doubles` to type double. We would need to check this below, but this will make things safer and prevent some issues with looping around in some severe low memory cases.